sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -1275,57 +1321,22 @@ def test_neighbors_badargs():
         nbrs.radius_neighbors_graph(X, mode="blah")
 
 
-def test_neighbors_metrics(n_samples=20, n_features=3, n_query_pts=2, n_neighbors=5):
+@pytest.mark.parametrize("metric", COMMON_VALID_METRICS)
```

Comment:
This is reducing the scope of the test too much (e.g. haversine is no longer tested).

Instead we should cover all of `neighbors.BallTree.valid_metrics` and keep the code handles the metrics that are expected to raise a `ValueError` for the `KDTree` subset and `continue`.